### PR TITLE
[test/plugin] immigrate test cassandra scenario.

### PR DESCRIPTION
**Pull request by aderm**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [ ] Bug fix
- [x] New feature provided
- [ ] Improve performance

- Related issues
#3583 
___
### Bug fix
- Bug description.

- How to fix?

___
### New feature or improvement
- Describe the details and related test reports.

### CODE REVIEW

File: test/plugin/scenarios/cassandra-java-driver-3.x-scenario/src/main/java/org/apache/skywalking/apm/testcase/cassandra/controller/CaseController.java [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    private static final String CREATE_TABLE_SQL = "CREATE TABLE IF NOT EXISTS demo.test(id TEXT PRIMARY KEY, value TEXT)";
+    private static final String INSERT_DATA_SQL = "INSERT INTO demo.test(id, value) VALUES(?,?)";
+    private static final String SELECT_DATA_SQL = "SELECT * FROM demo.test WHERE id = ?";
+    private static final String DELETE_DATA_SQL = "DELETE FROM demo.test WHERE id = ?";
+    private static final String DROP_TABLE_SQL = "DROP TABLE IF EXISTS demo.test";
+    private static final String DROP_KEYSPACE = "DROP KEYSPACE IF EXISTS demo";
+
+    private Cluster cluster;
+    private Session session;
+
```
Comment (arugal) [inline test/plugin/scenarios/cassandra-java-driver-3.x-scenario/src/main/java/org/apache/skywalking/apm/testcase/cassandra/controller/CaseController.java]: Hi, `cluster` and `session` should be a local variable, right?

**Comment by wu-sheng** [comment]
I think you break the YAML file, no test is running.

**Comment by aderm** [comment]
yes. a tab error

File: test/plugin/scenarios/cassandra-java-driver-3.x-scenario/pom.xml [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+                <exclusion>
+                    <artifactId>cassandra-driver-core</artifactId>
+                    <groupId>com.datastax.cassandra</groupId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>io.dropwizard.metrics</groupId>
+            <artifactId>metrics-jmx</artifactId>
+            <version>${metrics-jmx-version}</version>
```
Comment (arugal) [inline test/plugin/scenarios/cassandra-java-driver-3.x-scenario/pom.xml]: [maven repository](https://mvnrepository.com/artifact/io.dropwizard.metrics/metrics-jmx) doesn't seem to have this version

Comment (aderm) [reply test/plugin/scenarios/cassandra-java-driver-3.x-scenario/pom.xml]: Oops, yes, in my local environment, manually installed packages

Comment (aderm) [reply test/plugin/scenarios/cassandra-java-driver-3.x-scenario/pom.xml]: maven repository oldest version is 4.0.0, Let me try another version to see if it conflicts

File: test/plugin/scenarios/cassandra-java-driver-3.x-scenario/src/main/java/org/apache/skywalking/apm/testcase/cassandra/controller/CaseController.java (hunk 1 above)
Comment (aderm) [reply test/plugin/scenarios/cassandra-java-driver-3.x-scenario/src/main/java/org/apache/skywalking/apm/testcase/cassandra/controller/CaseController.java]: ACK

**Comment by wu-sheng** [comment]
> [2019-12-11 08:28:03:030] [ERROR] - org.apache.juli.logging.DirectJDKLog.log(DirectJDKLog.java:175) - Servlet.service() for servlet [dispatcherServlet] in context with path [/cassandra-java-driver-3.x-scenario] threw exception [Handler dispatch failed; nested exception is java.lang.NoClassDefFoundError: com/codahale/metrics/JmxReporter] with root cause

Your own codes are still using JMXReporter

**Comment by aderm** [comment]
config file setting is not effective. through `withoutJMXReporting()`  method is Ok.

File: docs/en/setup/service-agent/java-agent/Supported-list.md [hunk 3]
```diff
@@ -32,6 +32,7 @@
   * [Sharding-JDBC](https://github.com/shardingjdbc/sharding-jdbc) 1.5.x
   * [ShardingSphere](https://github.com/apache/incubator-shardingsphere) 3.0.0, 4.0.0-RC1
   * PostgreSQL Driver 8.x, 9.x, 42.x
+  * Cassandra Driver 3.7.0-3.7.2
```
Comment (dmsolr) [inline docs/en/setup/service-agent/java-agent/Supported-list.md]: Cassandra is already in the NoSQL section.

File: test/plugin/scenarios/cassandra-java-driver-3.x-scenario/src/main/java/org/apache/skywalking/apm/testcase/cassandra/controller/CaseController.java [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    @ResponseBody
+    public String testcase() {
+        logger.info("cassandra contact points: {}:{}", host, port);
+
+        Cluster cluster = null;
+        Session session = null;
+        try {
+            cluster = Cluster.builder().addContactPoint(host).withPort(port).withoutJMXReporting().build();
+            session = cluster.connect();
+            logger.info("cassandra connection open");
```
Comment (dmsolr) [inline test/plugin/scenarios/cassandra-java-driver-3.x-scenario/src/main/java/org/apache/skywalking/apm/testcase/cassandra/controller/CaseController.java]: The `Session` object was created in health-check. I think we can re-use it.

Comment (aderm) [reply test/plugin/scenarios/cassandra-java-driver-3.x-scenario/src/main/java/org/apache/skywalking/apm/testcase/cassandra/controller/CaseController.java]: healthCheck will execute many time in run.sh，so if you re-use，maybe session is not available.

File: docs/en/setup/service-agent/java-agent/Supported-list.md (hunk 3 above)
Comment (aderm) [reply docs/en/setup/service-agent/java-agent/Supported-list.md]: ACK

**Pull request merged**